tests/test-class-amp-dom-utils.php

```diff
@@ -145,4 +145,57 @@ public function test_html5_empty_elements() {
 		$this->assertEquals( 'div', $video->childNodes->item( 4 )->nodeName );
 		$this->assertEquals( 'span', $video->childNodes->item( 5 )->nodeName );
 	}
+
+	/**
+	 * Get Table Row Iterations
+	 *
+	 * @return array An array of arrays holding an integer representation of iterations.
+	 */
+	public function get_table_row_iterations() {
+		return array(
+			array( 1 ),
+			array( 10 ),
+			array( 100 ),
+			array( 1000 ),
+		);
+	}
+
+	/**
+	 * Tests attribute conversions on content with iframe srcdocs of variable lengths.
+	 *
+	 * @dataProvider get_table_row_iterations
+	 *
+	 * @param int $iterations The number of table rows to append to iframe srcdoc.
+	 */
+	public function test_attribute_conversion_on_long_iframe_srcdocs( $iterations ) {
+		$html = '<html amp><head><meta charset="utf-8"></head><body><table>';
+
+		for ( $i = 0; $i < $iterations; $i++ ) {
+			$html .= '
+				<tr>
+				<td class="rank" style="width:2%;">1453</td>
+				<td class="text" style="width:10%;">1947</td>
+				<td class="text">Pittsburgh Ironmen</td>
+				<td class="boolean" style="width:10%;text-align:center;"></td>
+				<td class="number" style="width:10%;">1242</td>
+				<td class="number">1192</td>
+				<td class="number">1111</td>
+				<td class="number highlight">1182</td>
+				</tr>
+			';
+		}
+
+		$html .= '</table></body></html>';
+
+		$to_convert = sprintf(
+			'<amp-iframe sandbox="allow-scripts" srcdoc="%s"> </amp-iframe>',
+			htmlentities( $html )
+		);
+
+		AMP_DOM_Utils::convert_amp_bind_attributes( $to_convert );
+
+		if ( preg_last_error() === PREG_BACKTRACK_LIMIT_ERROR ) {
+			$this->fail( 'Failed when backtrack limit was exhausted.' );
+		}
```

Comment:
When there is no error this test will run without any assertions. That will mean this test will appear as a warning when PHPUnit is called with ` --strict`.

I think it would be better to do:

```php
$this->assertSame( PREG_NO_ERROR, preg_last_error(), 'Probably failed when backtrack limit was exhausted.' );
```